update.sh

```diff
@@ -1,99 +1,219 @@
-        CLIENT_KEY_VAR=CLIENT_KEY_${CLUSTER}
-        # expand
-        CLIENT_CERT=${!CLIENT_CERT_VAR}
-        CLIENT_KEY=${!CLIENT_KEY_VAR}
-
-        if [[ ! -z "${CLIENT_CERT}" ]] && [[ ! -z "${CLIENT_KEY}" ]]; then
-          echo "[INFO] Setting client credentials with signed-certificate and key."
-          echo ${CLIENT_CERT} | base64 -d > client.crt
-          echo ${CLIENT_KEY} | base64 -d > client.key
-          kubectl config set-credentials ${USER} --client-certificate=client.crt --client-key=client.key
-        else
-          echo "[ERROR] Required plugin parameters:"
-          echo " - client_cert"
-          echo " - client_key"
-          echo "are not provided"
-          exit 1
-        fi
-      else
-        echo "[ERROR] Required plugin param - auth_mode - not provided"
-        echo "[INFO] Should be either [ token | client-cert ]"
-        exit 1
-      fi
+setGlobals(){
+  setUser
+  setNamespace
+  setDeployments
+  setContainers
+  setCluster
+  setServerUrl
+}
+
+setSecureCluster(){
+  local CLUSTER=$1; shift
+  local SERVER_URL=$1; shift
+  local SERVER_CERT=$1
+
+  echo "[INFO] Using secure connection with tls-certificate."
+  echo ${SERVER_CERT} | base64 -d > ca.crt
+  kubectl config set-cluster ${CLUSTER} --server=${SERVER_URL} --certificate-authority=ca.crt
+}
+
+setInsecureCluster(){
+  local CLUSTER=$1; shift
+  local SERVER_URL=$1
+
+  echo "[WARNING] Using insecure connection to cluster"
+  kubectl config set-cluster ${CLUSTER} --server=${SERVER_URL} --insecure-skip-tls-verify=true
+}
+
+setServerToken(){
+  local USER=$1; shift
+  local SERVER_TOKEN=$1
+
+  echo "[INFO] Setting client credentials with token"
+  kubectl config set-credentials ${USER} --token=${SERVER_TOKEN}
+}
+
+setClientCertAndKey(){
+  local USER=$1; shift
+  local CLIENT_CERT=$1; shift
+  local CLIENT_KEY=$1
+
+  echo "[INFO] Setting client credentials with signed-certificate and key."
+  echo ${CLIENT_CERT} | base64 -d > client.crt
+  echo ${CLIENT_KEY} | base64 -d > client.key
+  kubectl config set-credentials ${USER} --client-certificate=client.crt --client-key=client.key
+}
+
+setContext(){
+  local CLUSTER=$1; shift
+  local USER=$1
+
+  kubectl config set-context ${CLUSTER} --cluster=${CLUSTER} --user=${USER}
+  kubectl config use-context ${CLUSTER}
+}
+
+pollDeploymentRollout(){
+  local NAMESPACE=$1; shift
+  local DEPLOY=$1
+  # wait on deployment rollout status
+  while true; do
+    result=`kubectl -n ${NAMESPACE} rollout status --watch=false --revision=0 deployment/${DEPLOY}`
+    if [[ "${result}" == "deployment \"${DEPLOY}\" successfully rolled out" ]]; then
+      echo ${result}
+      exit 0
+    else
+      # TODO: more conditions for error handling based on result text
+      echo ${result}
+      sleep 10
     fi
+  done
+}
+
+startDeployment(){
+  local CLUSTER=$1; shift
+  local NAMESPACE=$1; shift
+  local DEPLOYMENTS=$1; shift
+  local CONTAINERS=$1
+
+  for DEPLOY in ${DEPLOYMENTS[@]}; do
```

Comment:
This does not seem to be running. We tried to deploy lumberjack [Ref](https://drone.razorpay.com/razorpay/lumberjack/38) to multiple deployments. but it skipped the 2 nd deployment `lumberjack-sqs`

Reply:
Thanks for the report.